### ISSUE TITLE
Unable to reproduce the error that leads to my released jar failing automated smoke test on Linux

### ISSUE THREAD

**Issue by Song-Mengfei** [nus-cs2103-AY2324S1/forum#196]
I received an email informing me that my released jar failed to pass the automated smoke test on Linux. The error message is:
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/122201960/d8022b21-f9a6-4d36-a0b5-9955ae4a1e00)

However, I am not able to reproduce this error on my friends's and my devices (both macOS and windows, ADM and ARM chips) when run with java 11.

This is the link to my release page: https://github.com/Song-Mengfei/ip/releases

This is my build.gradle file:
```
plugins {
    id 'java'
    id 'application'
    id 'checkstyle'
    id 'org.openjfx.javafxplugin' version '0.0.13'
    id 'com.github.johnrengelman.shadow' version '7.1.2'
}

checkstyle {
    toolVersion = '10.2'
}

checkstyleMain {
    source 'src/main/java'
}

repositories {
    mavenCentral()
}

dependencies {
    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0'
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'

    String javaFxVersion = '17.0.7'

    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'
}

test {
    useJUnitPlatform()

    testLogging {
        events "passed", "skipped", "failed"

        showExceptions true
        exceptionFormat "full"
        showCauses true
        showStackTraces true
        showStandardStreams = false
    }
}

application {
    mainClass.set('duke.Launcher')
}

shadowJar {
    archiveBaseName = "duke"
    archiveClassifier = null
    dependsOn("distZip", "distTar")
}

javafx {
    version = '17.0.2'
    modules = [ 'javafx.controls' , 'javafx.fxml' ]
}

run{
    enableAssertions = true
    standardInput = System.in
}
```
Thanks.

**Comment by damithc** [comment]
@Song-Mengfei While waiting for inputs from others, you may want to use [fenced code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to format your post for better readability.

**Comment by ryamgoh** [comment]
same here. i'm getting a similar error
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/122634188/1d6a7b88-02f5-48f3-9171-c151a3513273)

**Comment by woojiahao** [comment]
If you used the JavaFX plugin, might I suggest trying out this fix: #173 I've tested it extensively on Linux x86 architecture and it runs

**Comment by damithc** [comment]
Perhaps a few other Linux users can check the affected JAR files to see if they can reproduce the error?

**Comment by seanpzk** [comment]
> If you used the JavaFX plugin, might I suggest trying out this fix: #173 I've tested it extensively on Linux x86 architecture and it runs

I am facing similar issues, I actually tried this solution and I am trying to test [duke.zip](https://github.com/nus-cs2103-AY2324S1/forum/files/12705951/duke.zip) using VirtualBox. Unfortunately, I am not sure if this is the same issue or is it a problem with my installation with the virtual machine. Would appreciate if any Linux users can help check! Thank you

![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/122248828/acfb15d8-2d5a-432f-9b27-1041fabe39b7)

**Comment by woojiahao** [comment]
This errors seems to be occurring because your installation of Ubuntu does not support GUIs, it's a separate issue. On Ubuntu, it might be due to not having the entire OS installed on the VM seeing that you're still on `/home/ubuntu`.

**Comment by Song-Mengfei** [comment]
> If you used the JavaFX plugin, might I suggest trying out this fix: #173 I've tested it extensively on Linux x86 architecture and it runs

I used your solution to fix the error. It works well on both windows and MacOS, can you help me to test my jar on your Linux enviornment ?
This is my jar: https://github.com/Song-Mengfei/ip/releases/tag/A-NewJar
Thank you a lot!

**Comment by lawruixi** [comment]
@Song-Mengfei Heya, I tried your JAR on my Linux machine and it runs perfectly. :D
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/52446119/88129b4c-68d3-4320-be1f-6f77b12e1d0f)

**Comment by Song-Mengfei** [comment]
> @Song-Mengfei Heya, I tried your JAR on my Linux machine and it runs perfectly. :D ![image](https://user-images.githubusercontent.com/52446119/270112001-88129b4c-68d3-4320-be1f-6f77b12e1d0f.png)

Thank you!

**Comment by ryamgoh** [comment]
Hi, thank you so much for the help @woojiahao !!
Your solution works :)

**Comment by zacwong2151** [comment]
> > If you used the JavaFX plugin, might I suggest trying out this fix: #173 I've tested it extensively on Linux x86 architecture and it runs
> 
> I am facing similar issues, I actually tried this solution and I am trying to test [duke.zip](https://github.com/nus-cs2103-AY2324S1/forum/files/12705951/duke.zip) using VirtualBox. Unfortunately, I am not sure if this is the same issue or is it a problem with my installation with the virtual machine. Would appreciate if any Linux users can help check! Thank you
> 
> ![image](https://user-images.githubusercontent.com/122248828/270096697-acfb15d8-2d5a-432f-9b27-1041fabe39b7.png)

Hi, I'm facing the same issue as @seanpzk, where I followed @woojiahao advice and now have the following error message:

![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/111118549/2cacb02e-ecde-4b2e-b722-f7ac2d547579)

I believe I have the entire Linux OS installed tho. Anyone able to help me on this issue? Below is a link to my jar release.

https://github.com/zacwong2151/ip/releases/tag/v0.2

Also below is the relevant portion of my build.gradle file in case it is necessary 

```
plugins {
    id 'java'
    id 'application'
    id 'checkstyle'
    id 'com.github.johnrengelman.shadow' version '7.1.2'
    id 'org.openjfx.javafxplugin' version '0.0.13'
}

repositories {
    mavenCentral()
}

javafx {
    version = '17.0.7'
    modules = ['javafx.base', 'javafx.controls', 'javafx.fxml', 'javafx.graphics']
}

dependencies {
    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0'
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'
    implementation group: 'com.google.code.gson', name: 'gson', version: '2.10.1'

    // Used for loops to clean up the imports
    def platforms = ["win", "linux", "mac"]
    def javafxDependency = ["javafx-graphics", "javafx-controls", "javafx-fxml"]

    for (plt in platforms) {
        for (dep in javafxDependency) {
            runtimeOnly "org.openjfx:$dep:$javafx.version:$plt"
        }
    }
}
```

**Comment by woojiahao** [comment]
It looks like you're using WSL, I believe WSL doesn't support GUIs
